src/data/set/enumerate.lean

```diff
@@ -2,34 +2,40 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
-
-Enumerate elements of a set with a select function.
 -/
 import data.set.lattice
 import tactic.wlog
+
+/-!
+# Set enumeration
+
+This file allows enumeration of sets given a choice function.
+-/
+
 noncomputable theory
 
-open function set
+open function
 
 namespace set
 
 section enumerate
 parameters {α : Type*} (sel : set α → option α)
 
+/-- Given a choice function `sel`, enumerates the elements of a set. -/
```

Comment:
It would be nice to understand what `enumerate` actually does without reading the code, so:
```suggestion
/-- Given a choice function `sel`, enumerates the elements of a set in the order
`a 0 = sel s`, `a 1 = sel (s \ {a 0})`, `a 2 = sel (s \ {a 0, a 1})`, ...;
stopping when `sel (s \ {a 0, ..., a n}) = none`.

The definition does not assume `sel` actually is a choice function, i.e. `sel s ∈ s` and `sel s = none ↔ s = ∅`;
these assumptions can be added to lemmas as needed. 
-/
```